data/mods/gen2stadium/scripts.ts

```diff
@@ -0,0 +1,555 @@
+				damage += Math.floor(damage / 2);
+			}
+
+			// Type effectiveness
+			const totalTypeMod = target.runEffectiveness(move);
+			// Super effective attack
+			if (totalTypeMod > 0) {
+				if (!suppressMessages) this.battle.add('-supereffective', target);
+				damage *= 2;
+				if (totalTypeMod >= 2) {
+					damage *= 2;
+				}
+			}
+			// Resisted attack
+			if (totalTypeMod < 0) {
+				if (!suppressMessages) this.battle.add('-resisted', target);
+				damage = Math.floor(damage / 2);
+				if (totalTypeMod <= -2) {
+					damage = Math.floor(damage / 2);
+				}
+			}
+
+			// Apply random factor is damage is greater than 1, except for Flail and Reversal
+			if (!move.noDamageVariance && damage > 1) {
+				damage *= this.battle.random(217, 256);
+				damage = Math.floor(damage / 255);
+			}
+
+			// If damage is less than 1, we return 1
+			if (basePower && !Math.floor(damage)) {
+				return 1;
+			}
+
+			// We are done, this is the final damage
+			return damage;
+		},
+	},
+	//Stadium 2 ignores stat drops due to status ailments upon boosting the dropped stat.
+	//For example: if a burned Snorlax uses Curse then it will ignore the attack drop from burn when it is recalculating its attack stat.
+	//This is why volatiles are added to status conditions, so that we can keep track of whether or not to apply the stat drop from statuses.
+	boost(boost, target, source = null, effect = null) {
+		if (this.event) {
+			if (!target) target = this.event.target;
+			if (!source) source = this.event.source;
+			if (!effect) effect = this.effect;
+		}
+		if (typeof effect === 'string') effect = this.dex.conditions.get(effect);
+		if (!target?.hp) return 0;
+		let success = null;
+		boost = this.runEvent('Boost', target, source, effect, {...boost});
+		let i: BoostID;
+		for (i in boost) {
+			const currentBoost: SparseBoostsTable = {};
+			currentBoost[i] = boost[i];
+			if (boost[i] !== 0 && target.boostBy(currentBoost)) {
+				success = true;
+				let msg = '-boost';
+				// Check for boost increases deleting attack or speed drops
+				if (i === 'atk' && target.status === 'brn' && target.volatiles['brnattackdrop']) {
+					target.removeVolatile('brnattackdrop');
+				}
+				if (i === 'spe' && target.status === 'par' && target.volatiles['parspeeddrop']) {
+					target.removeVolatile('parspeeddrop');
+				}
+				if (!effect || effect.effectType === 'Move') {
+					this.add(msg, target, i, boost[i]);
+				} else {
+					this.add(msg, target, i, boost[i], '[from] ' + effect.fullname);
+				}
+				this.runEvent('AfterEachBoost', target, source, effect, currentBoost);
+			}
+		}
+		this.runEvent('AfterBoost', target, source, effect, boost);
+		return success;
+	},
+	/**
+	 * Implementing Self-KO Clause by having it check what the last move used by the players were in the case both Pokemon faint.
+	 * Since the only way this can happen in Stadium 2 is if a player uses self-destruct or explosion, I can use this to determine who should win.
```

Comment:
```suggestion
	 * Implementing Self-KO Clause by having it check what the last move used by the players were
	 * in the case both Pokemon faint. Since the only way this can happen in Stadium 2 is if a player
	 * uses self-destruct or explosion, I can use this to determine who should win.
```